client/signup/steps/new-or-existing-site/index.tsx

```diff
@@ -0,0 +1,58 @@
+import { useTranslate } from 'i18n-calypso';
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import StepWrapper from 'calypso/signup/step-wrapper';
+import { removeSiteSlugDependency } from 'calypso/state/signup/actions';
+import { saveSignupStep, submitSignupStep } from 'calypso/state/signup/progress/actions';
+import NewOrExistingSiteScreen from './new-or-existing-site';
+import { ChoiceType } from './types';
+
+interface Props {
+	goToNextStep: () => void;
+	submitSignupStep: ( { stepName, wasSkipped }: { stepName: string; wasSkipped: boolean } ) => void;
+	goToStep: ( stepName: string ) => void;
+	stepName: string;
+}
+
+export default function NewOrExistingSiteStep( props: Props ): React.ReactNode {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+
+	const headerText = translate( 'Choose where you want us to build your site.' );
+	const subHeaderText = '';
+
+	useEffect( () => {
+		dispatch( saveSignupStep( { stepName: props.stepName } ) );
```

Comment:
Why do we save the signup step here?

Reply:
We discussed this briefly in Slack, but I'll add some context here for anyone searching. `saveSignupStep` adds the step to the `signup.progress` slice of the redux store. Saving the step is needed for back functionality to work correctly, especially when skipping a step is allowed.

In this particular case, the only way to move forward from this step is to choose either of the two options (New/Existing site), which calls `submitSignupStep`. `submitSignupStep` also saves the step to the redux store, so calling `saveSignupStep` here is not necessary. However, if we move this step to a different flow or allow skipping this step, we would require `saveSignupStep` here.